java/core/src/main/java/com/lancedb/lance/Dataset.java

```diff
@@ -304,6 +305,38 @@ public void addColumns(ArrowArrayStream stream, Optional<Long> batchSize) {
   private native void nativeAddColumnsByReader(
       long arrowStreamMemoryAddress, Optional<Long> batchSize);
 
+  /**
+   * Add columns to the dataset.
+   *
+   * @param schema The Arrow schema definitions to add columns.
+   * @param batchSize The number of rows to read at a time from the source dataset when applying the
+   *     transform.
+   */
+  public void addColumns(Schema schema, Optional<Long> batchSize) {
```

Comment:
Can we show what the difference is between a batch size that is empty or non-empty?